src/lib.rs

```diff
@@ -78,18 +78,36 @@ impl<R: Runtime> Plugin<R> for WindowState {
 
         let cache = self.cache.clone();
         let label = window.label().to_string();
+        let window_clone = window.clone();
         window.on_window_event(move |e| match e {
             WindowEvent::Moved(position) => {
-                let mut c = cache.lock().unwrap();
-                let state = c.get_mut(&label).unwrap();
-                state.x = position.x;
-                state.y = position.y;
+                let size = window_clone.inner_size().unwrap();
+                // On some platforms, when a window gets minimized, it will report a
+                // position with a huge negative value (-32000), so for now
+                // don't save the position if the window is not visible on screen.
+                // also subtract a 25px, just to ensure there is enough space to
+                // be able to resize the window.
+                // TODO: use `window.is_minimized()` once it is implemented
+                if position.x > 0 - size.width as i32 - 25
```

Comment:
Hmm this will conflict with multi-monitor setups i think. For example (on windows) if i move the window to the monitor on the left side of the primary one the reported position is `x: -1567, y: 171`.

but in general, that's not the only multi monitor problem we have (for example if you unplug a monitor), sooo idk.

Reply:
I had multi-monitor in mind when I wrote this but assumed that any `0,0` is the coordinates of the left-most monitor

Reply:
I guess I can get the monitor handle and adjust the logic accordingly

Reply:
I think linux starts at 0,0. Or at least x does. But y can still be negative iirc.

Reply:
I don't have a multi monitor setup, can you test if it works ?

Reply:
Works perfect on windows 👍 played around with some weird monitor positioning (including x **and** y), all good (well of course with the unpluggin thing).
I probably can test linux later today too.

Reply:
thanks, appreciate your help

Reply:
Hmm, linux (wayland) doesn't work, because the Moved event always reports a **window** position of 0,0. But the monitor sizes/positions never get below 0 so i think it should work once the window position works again :thinking:

Reply:
does the moved event work correctly on wayland in tao or in tauri events ?

Reply:
it's broken in tao too and it looks like i have to trigger another event to get the moved event fired. Speaking of which, it also looks like the moved event gets triggered on every other window event too (for example resizing and blur/focus triggers it too

Reply:
Alright, I will consider this PR is working well and ready to go. please open an issue with your findings on tao for the moved event.